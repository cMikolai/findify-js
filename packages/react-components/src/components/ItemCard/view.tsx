import React from 'react'

export default ({ item, theme }) => (
  <div className={theme.root}>
    <h4 className={theme.title}>{ item.get('title') }</h4>
    <img className={theme.image} src={item.get('image_url')} />
  </div>
);
