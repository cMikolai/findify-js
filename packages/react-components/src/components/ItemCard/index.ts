import { defaultProps } from 'recompose';
import view from './view'
import theme from './styles.css';

console.log(defaultProps);

export default defaultProps({ theme })(view);
