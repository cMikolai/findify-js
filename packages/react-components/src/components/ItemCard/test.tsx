import React from 'react'
import { shallow } from 'enzyme'
import ItemCard from './index'
import itemMock from '../../../jest/item.mock';
describe('ItemCard', () => {
  it('renders correctly', () => {
    expect(shallow(<ItemCard item={itemMock(
      'Test Title',
      'https://picsum.photos/100/200'
    )} />)).toMatchSnapshot()
  })
})
