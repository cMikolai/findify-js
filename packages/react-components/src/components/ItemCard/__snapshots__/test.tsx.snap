// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<Component
  item={
    Object {
      "get": [Function],
      "hashCode": [Function],
      "image_url": "https://picsum.photos/100/200",
      "title": "Test Title",
    }
  }
  theme={null}
/>
`;
